code/src/test_usart.cpp

```diff
@@ -0,0 +1,57 @@
+#include <test_usart.hpp>
+
+namespace r2d2 {
+
+    test_usart_c::test_usart_c(){}
+
+    void test_usart_c::enable(){
+        //Not needed in test implementation
+    }
+
+    void test_usart_c::disable(){
+        //Not needed in test implementation
+    }
+
+    bool test_usart_c::send(const uint8_t c) {
+        return true;
+        //Not needed in test implementation
+    }
+
+    void test_usart_c::putc(char c) {
+        send(c);
+    }
+
+    uint8_t test_usart_c::receive() {
+        return receive_buffer.copy_and_pop();
+    }
+
+    bool test_usart_c::char_available() {
+        return !receive_buffer.empty();
+    }
+
+    char test_usart_c::getc() {
+        return receive();
+    }
+
+    unsigned int test_usart_c::available() {
+        return !receive_buffer.empty();
+    }
+
+    void test_usart_c::set_receive_string(const std::string &str) {
+        receive_buffer.clear();
+        for(char c : str){
+            set_receive_byte(c);
+        }
+    }
+
+    void test_usart_c::set_receive_bytes(const std::vector<uint8_t> &bytes){
+        receive_buffer.clear();
```

Comment:
why are you clearing the receive_buffer? what if i want to add some bytes first with the set_receive_byte function